### ISSUE TITLE
What are some of your go-to resources for CSS architecture? 

### ISSUE THREAD

**Issue by vkraucunas** [jackrugile/ama#3]
I'm familiar with Bourbon + Bitters, but that's about as far as I've gotten on the topic.

**Comment by jackrugile** [comment]
This question makes me sad because I don't have a good answer. And I really really really should have a good answer.

Honestly, even with somewhat smaller projects, organizing things, naming things, and having any sort of strategy has always been a struggle for me. I usually keep a roughly similar organization of files in Sass (that's what I usually use), and then I'll write from scratch as I need things. I of course will pull code from similar components used in other projects here and there as needed too.

I work on a pretty small dev team. We typically see projects from start to finish individually, so we can get away with some bad personal habits. It's only when someone else goes to touch our code that everything crumbles and you have to answer why you did something the way you did. And damn it when that happens!

> "Don't make me explain my code, bro. I'm a free spirit and I do what I want. Plus, it's experimental and abstract, you wouldn't understand."

This is starting to shift lately though. We're collaborating on more projects these days. Since inconsistency in strategy and in style is becoming an issue, we've been having discussions about ways to get organized. So many questions come up like:
- What files? How many files? In what directories?
- How do we order our CSS rules?
- How do we make things reusable without making them so locked together that changing one thing can have nasty unintentional side effects across a project?
- How can we reduce specificity issues?
- How much should we really be nesting our selectors and rules?
- Function? Mixin? Extend class? Extend placeholder? Hardcode that shit?
- Keep our markup clean or keep our styles clean? Can we have both?
- SMACSS? OOCSS? ACSS? BEM? One? Some? All?
- How should we go about naming our variables?

## 

And I honestly don't know the answers to most of those questions. So, in a way I am grateful that I know that I know nothing, but it still sucks because, ya know, I know nothing. Also, I realize some of those questions above go beyond what you were even asking anyway.

To sum up, I'm lost, and if you have any suggestions, I'm all ears. :ear::ear::ear: I'll check out Bourbon + Bitters to start. I know those don't solve ALL my problems, but it might solve some.

**Comment by lualparedes** [comment]
I researched this topic thoroughly during last year and I ended up creating a mini-guide on the topic: 

- https://docs.google.com/document/d/14LGa6KMBvZrY0bp1mhxyhfA1r5auXboX7aNHFW4oE6Q/edit?usp=sharing

 and I merged what I liked the most into my own methodology: 

- https://github.com/lualparedes/frontend-methodology/blob/master/css.md
 
Just my two cents, hope somebody finds this useful! :blush:
